rabbitmq/src/main/java/io/micronaut/rabbitmq/intercept/RabbitMQConsumerAdvice.java

```diff
@@ -486,7 +486,8 @@ public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProp
                     // memory and will call handleDelivery for these messages, even if they are re-queued by the broker.
                     // The client will be unable to acknowledge these messages. So it is safe to silently discard
                     // them, without bothering the callback handler.
-                    if (!getChannel().isOpen()) {
```

Comment:
There's already a chunk of code in the RecoverableConsumerWrapper.cancel() method, which upon receiving the SIGTERM signal, makes sure to finish the processing for the message[s] that have been already delivered to the consumer threads. With this fix, the rest of the queued messages are not going to be pulled from the queue after receiving the SIGTERM signal.